plugins/by-name/lsp-format/default.nix

```diff
@@ -89,35 +89,28 @@ in
     };
   };
 
-  config =
-    let
-      setupOptions = cfg.setup // cfg.extraOptions;
-    in
-    mkIf cfg.enable {
-      warnings = mkIf (!config.plugins.lsp.enable) [
-        "You have enabled `plugins.lsp-format` but have `plugins.lsp` disabled."
-      ];
-
-      extraPlugins = [ cfg.package ];
+  extraConfig = cfg: {
+    warnings = lib.mkIf (!config.plugins.lsp.enable) [
+      "You have enabled `plugins.lsp-format` but have `plugins.lsp` disabled."
+    ];
 
-      plugins.lsp = {
-        onAttach = mkIf (cfg.lspServersToEnable == "all") ''
-          require("lsp-format").on_attach(client)
-        '';
+    plugins.lsp = {
+      onAttach =
+        lib.mkIf (cfg.lspServersToEnable == "all") # Lua
+          ''
+            require("lsp-format").on_attach(client)
+          '';
 
-        servers =
-          if (isList cfg.lspServersToEnable) then
-            genAttrs cfg.lspServersToEnable (serverName: {
-              onAttach.function = ''
+      servers =
+        if (lib.isList cfg.lspServersToEnable) then
```

Comment:
Shouldn't we use `lib.optionalAttrs` here ?

Reply:
This is just a formatting change (the `if then else` is pre-existing code), but yes it can be simplified to `optionalAttrs`.

Maybe it should even be a `mkIf`, I'm not sure... (they do subtly different things)

Reply:
Yes, this was pre-existing code.



> Maybe it should even be a `mkIf`, I'm not sure... (they do subtly different things)

`optionalAttrs` and `if ... then ... else {}` are equivalent. However `mkIf` operates at a later stage (when the option values are resolved).
`mkIf` should be better then.

Reply:
optionalAttrs works but for some reason mkIf fails because it still tries to evaluate the inner expression with the default string value.

Reply:
Yeah, one of the advertised advantages(?) of `mkIf` is that the module system _usually_ still evaluates the definition value, even when the condition is false.

Thanks for testing it out!

Reply:
Ah yeah, I remember us talking about that before. Makes sense.